File: app/src/main/java/com/example/basic/WeeklySchedule.kt
package com.example.basic

// Data classes and sample schedule used for PlannerScreen

data class ClassEntry(
    val course: String,
    val faculty: String,
    val start: String,
    val end: String,
    val room: String
)

val WEEKLY_SCHEDULE: Map<String, List<ClassEntry>> = mapOf(
    "Monday" to listOf(
        ClassEntry("CSE1001", "Prof. Rao", "08:00", "08:50", "101"),
        ClassEntry("MAT1002", "Dr. Singh", "09:00", "09:50", "201"),
        ClassEntry("PHY1003", "Dr. Patel", "10:00", "10:50", "Lab 1"),
        ClassEntry("ENG1004", "Ms. James", "11:00", "11:50", "305"),
        ClassEntry("CHE1005", "Dr. Verma", "12:00", "12:50", "202"),
        ClassEntry("CSE2001", "Prof. Gupta", "13:00", "13:50", "102"),
        ClassEntry("MAT2002", "Dr. Mehta", "14:00", "14:50", "203"),
        ClassEntry("PHY2003", "Dr. Shah", "15:00", "15:50", "Lab 2"),
        ClassEntry("ENG2004", "Ms. Clark", "16:00", "16:50", "306")
    ),
    "Tuesday" to listOf(
        ClassEntry("CHE1005", "Dr. Verma", "08:00", "08:50", "202"),
        ClassEntry("CSE1001", "Prof. Rao", "09:00", "09:50", "101"),
        ClassEntry("MAT1002", "Dr. Singh", "10:00", "10:50", "201"),
        ClassEntry("PHY1003", "Dr. Patel", "11:00", "11:50", "Lab 1"),
        ClassEntry("CHE2005", "Dr. Jain", "12:00", "12:50", "204"),
        ClassEntry("CSE2001", "Prof. Gupta", "13:00", "13:50", "102"),
        ClassEntry("MAT2002", "Dr. Mehta", "14:00", "14:50", "203"),
        ClassEntry("PHY2003", "Dr. Shah", "15:00", "15:50", "Lab 2"),
        ClassEntry("ENG2004", "Ms. Clark", "16:00", "16:50", "306")
    ),
    "Wednesday" to listOf(
        ClassEntry("ENG1004", "Ms. James", "08:00", "08:50", "305"),
        ClassEntry("CSE1001", "Prof. Rao", "09:00", "09:50", "101"),
        ClassEntry("CHE1005", "Dr. Verma", "10:00", "10:50", "202"),
        ClassEntry("MAT1002", "Dr. Singh", "11:00", "11:50", "201"),
        ClassEntry("ELE2001", "Dr. Nair", "12:00", "12:50", "401"),
        ClassEntry("CSE2001", "Prof. Gupta", "13:00", "13:50", "102"),
        ClassEntry("MAT2002", "Dr. Mehta", "14:00", "14:50", "203"),
        ClassEntry("PHY2003", "Dr. Shah", "15:00", "15:50", "Lab 2"),
        ClassEntry("ENG2004", "Ms. Clark", "16:00", "16:50", "306")
    ),
    "Thursday" to listOf(
        ClassEntry("PHY1003", "Dr. Patel", "08:00", "08:50", "Lab 1"),
        ClassEntry("ENG1004", "Ms. James", "09:00", "09:50", "305"),
        ClassEntry("CSE1001", "Prof. Rao", "10:00", "10:50", "101"),
        ClassEntry("CHE1005", "Dr. Verma", "11:00", "11:50", "202"),
        ClassEntry("PHY2003", "Dr. Shah", "12:00", "12:50", "Lab 2"),
        ClassEntry("CSE2001", "Prof. Gupta", "13:00", "13:50", "102"),
        ClassEntry("MAT2002", "Dr. Mehta", "14:00", "14:50", "203"),
        ClassEntry("ENG2004", "Ms. Clark", "15:00", "15:50", "306"),
        ClassEntry("CHE2005", "Dr. Jain", "16:00", "16:50", "204")
    ),
    "Friday" to listOf(
        ClassEntry("MAT1002", "Dr. Singh", "08:00", "08:50", "201"),
        ClassEntry("PHY1003", "Dr. Patel", "09:00", "09:50", "Lab 1"),
        ClassEntry("ENG1004", "Ms. James", "10:00", "10:50", "305"),
        ClassEntry("CHE1005", "Dr. Verma", "11:00", "11:50", "202"),
        ClassEntry("CSE2001", "Prof. Gupta", "12:00", "12:50", "102"),
        ClassEntry("MAT2002", "Dr. Mehta", "13:00", "13:50", "203"),
        ClassEntry("PHY2003", "Dr. Shah", "14:00", "14:50", "Lab 2"),
        ClassEntry("ENG2004", "Ms. Clark", "15:00", "15:50", "306"),
        ClassEntry("CHE2005", "Dr. Jain", "16:00", "16:50", "204")
    ),
    "Saturday" to listOf(
        ClassEntry("CSE1001", "Prof. Rao", "09:00", "10:30", "101"),
        ClassEntry("LAB Project", "Staff", "10:45", "12:15", "Innovation Lab"),
        ClassEntry("MAT1002", "Dr. Singh", "12:30", "13:20", "201"),
        ClassEntry("ELE2001", "Dr. Nair", "13:30", "14:20", "401"),
        ClassEntry("CSE2001", "Prof. Gupta", "14:30", "15:20", "102"),
        ClassEntry("MAT2002", "Dr. Mehta", "15:30", "16:20", "203"),
        ClassEntry("PHY2003", "Dr. Shah", "16:30", "17:20", "Lab 2"),
        ClassEntry("ENG2004", "Ms. Clark", "17:30", "18:20", "306"),
        ClassEntry("CHE2005", "Dr. Jain", "18:30", "19:20", "204")
    )
)

File: vit-student-app/src/data/weeklySchedule.ts
export type ClassEntry = {
  course: string;
  faculty: string;
  start: string;
  end: string;
  room: string;
};

export type WeeklySchedule = {
  [day: string]: ClassEntry[];
};

export const WEEKLY_SCHEDULE: WeeklySchedule = {
  Monday: [
    { course: 'CSE1001', faculty: 'Prof. Rao', start: '08:00', end: '08:50', room: '101' },
    { course: 'MAT1002', faculty: 'Dr. Singh', start: '09:00', end: '09:50', room: '201' },
    { course: 'PHY1003', faculty: 'Dr. Patel', start: '10:00', end: '10:50', room: 'Lab 1' },
    { course: 'ENG1004', faculty: 'Ms. James', start: '11:00', end: '11:50', room: '305' },
    { course: 'CHE1005', faculty: 'Dr. Verma', start: '12:00', end: '12:50', room: '202' },
    { course: 'CSE2001', faculty: 'Prof. Gupta', start: '13:00', end: '13:50', room: '102' },
    { course: 'MAT2002', faculty: 'Dr. Mehta', start: '14:00', end: '14:50', room: '203' },
    { course: 'PHY2003', faculty: 'Dr. Shah', start: '15:00', end: '15:50', room: 'Lab 2' },
    { course: 'ENG2004', faculty: 'Ms. Clark', start: '16:00', end: '16:50', room: '306' },
  ],
  Tuesday: [
    { course: 'CHE1005', faculty: 'Dr. Verma', start: '08:00', end: '08:50', room: '202' },
    { course: 'CSE1001', faculty: 'Prof. Rao', start: '09:00', end: '09:50', room: '101' },
    { course: 'MAT1002', faculty: 'Dr. Singh', start: '10:00', end: '10:50', room: '201' },
    { course: 'PHY1003', faculty: 'Dr. Patel', start: '11:00', end: '11:50', room: 'Lab 1' },
    { course: 'CHE2005', faculty: 'Dr. Jain', start: '12:00', end: '12:50', room: '204' },
    { course: 'CSE2001', faculty: 'Prof. Gupta', start: '13:00', end: '13:50', room: '102' },
    { course: 'MAT2002', faculty: 'Dr. Mehta', start: '14:00', end: '14:50', room: '203' },
    { course: 'PHY2003', faculty: 'Dr. Shah', start: '15:00', end: '15:50', room: 'Lab 2' },
    { course: 'ENG2004', faculty: 'Ms. Clark', start: '16:00', end: '16:50', room: '306' },
  ],
  Wednesday: [
    { course: 'ENG1004', faculty: 'Ms. James', start: '08:00', end: '08:50', room: '305' },
    { course: 'CSE1001', faculty: 'Prof. Rao', start: '09:00', end: '09:50', room: '101' },
    { course: 'CHE1005', faculty: 'Dr. Verma', start: '10:00', end: '10:50', room: '202' },
    { course: 'MAT1002', faculty: 'Dr. Singh', start: '11:00', end: '11:50', room: '201' },
    { course: 'ELE2001', faculty: 'Dr. Nair', start: '12:00', end: '12:50', room: '401' },
    { course: 'CSE2001', faculty: 'Prof. Gupta', start: '13:00', end: '13:50', room: '102' },
    { course: 'MAT2002', faculty: 'Dr. Mehta', start: '14:00', end: '14:50', room: '203' },
    { course: 'PHY2003', faculty: 'Dr. Shah', start: '15:00', end: '15:50', room: 'Lab 2' },
    { course: 'ENG2004', faculty: 'Ms. Clark', start: '16:00', end: '16:50', room: '306' },
  ],
  Thursday: [
    { course: 'PHY1003', faculty: 'Dr. Patel', start: '08:00', end: '08:50', room: 'Lab 1' },
    { course: 'ENG1004', faculty: 'Ms. James', start: '09:00', end: '09:50', room: '305' },
    { course: 'CSE1001', faculty: 'Prof. Rao', start: '10:00', end: '10:50', room: '101' },
    { course: 'CHE1005', faculty: 'Dr. Verma', start: '11:00', end: '11:50', room: '202' },
    { course: 'PHY2003', faculty: 'Dr. Shah', start: '12:00', end: '12:50', room: 'Lab 2' },
    { course: 'CSE2001', faculty: 'Prof. Gupta', start: '13:00', end: '13:50', room: '102' },
    { course: 'MAT2002', faculty: 'Dr. Mehta', start: '14:00', end: '14:50', room: '203' },
    { course: 'ENG2004', faculty: 'Ms. Clark', start: '15:00', end: '15:50', room: '306' },
    { course: 'CHE2005', faculty: 'Dr. Jain', start: '16:00', end: '16:50', room: '204' },
  ],
  Friday: [
    { course: 'MAT1002', faculty: 'Dr. Singh', start: '08:00', end: '08:50', room: '201' },
    { course: 'PHY1003', faculty: 'Dr. Patel', start: '09:00', end: '09:50', room: 'Lab 1' },
    { course: 'ENG1004', faculty: 'Ms. James', start: '10:00', end: '10:50', room: '305' },
    { course: 'CHE1005', faculty: 'Dr. Verma', start: '11:00', end: '11:50', room: '202' },
    { course: 'CSE2001', faculty: 'Prof. Gupta', start: '12:00', end: '12:50', room: '102' },
    { course: 'MAT2002', faculty: 'Dr. Mehta', start: '13:00', end: '13:50', room: '203' },
    { course: 'PHY2003', faculty: 'Dr. Shah', start: '14:00', end: '14:50', room: 'Lab 2' },
    { course: 'ENG2004', faculty: 'Ms. Clark', start: '15:00', end: '15:50', room: '306' },
    { course: 'CHE2005', faculty: 'Dr. Jain', start: '16:00', end: '16:50', room: '204' },
  ],
  Saturday: [
    { course: 'CSE1001', faculty: 'Prof. Rao', start: '09:00', end: '10:30', room: '101' },
    { course: 'LAB Project', faculty: 'Staff', start: '10:45', end: '12:15', room: 'Innovation Lab' },
    { course: 'MAT1002', faculty: 'Dr. Singh', start: '12:30', end: '13:20', room: '201' },
    { course: 'ELE2001', faculty: 'Dr. Nair', start: '13:30', end: '14:20', room: '401' },
    { course: 'CSE2001', faculty: 'Prof. Gupta', start: '14:30', end: '15:20', room: '102' },
    { course: 'MAT2002', faculty: 'Dr. Mehta', start: '15:30', end: '16:20', room: '203' },
    { course: 'PHY2003', faculty: 'Dr. Shah', start: '16:30', end: '17:20', room: 'Lab 2' },
    { course: 'ENG2004', faculty: 'Ms. Clark', start: '17:30', end: '18:20', room: '306' },
    { course: 'CHE2005', faculty: 'Dr. Jain', start: '18:30', end: '19:20', room: '204' },
  ],
};

export default WEEKLY_SCHEDULE;
